server/py/services/api/crud/runtimes/nuclio/function.py

```diff
@@ -418,11 +418,24 @@ def _configure_serving_spec(
                     + "in the returned json, and check if the function runs successfully. "
                     + "Repeat as necessary to get the spec to an allowed size"
                 )
-            # Compress and encode the serving spec
-            compressed_serving_spec = gzip.compress(serving_spec.encode("utf-8"))
-            encoded_serving_spec = base64.b64encode(compressed_serving_spec).decode(
-                "utf-8"
-            )
+            if (
+                not client_version
+                or semver.Version.parse(client_version)
+                >= semver.Version.parse("1.8.0-rc20")
+                or "unstable" in client_version
+            ):
+                # Compress and encode the serving spec
+                compressed_serving_spec = gzip.compress(serving_spec.encode("utf-8"))
+                encoded_serving_spec = base64.b64encode(compressed_serving_spec).decode(
+                    "utf-8"
+                )
+            else:
+                # TODO: remove in 1.10.0.
```

Comment:
in this case the config-map limit is 1MB, so I suggest you move the checking of serving_spec_len to be tailored for the zipped/non-zipped case.